File: app/src/main/java/com/luisenricke/simpleroomapp/database/converter/DateConverter.java
package com.luisenricke.simpleroomapp.database.converter;

import android.annotation.SuppressLint;

import androidx.room.TypeConverter;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.time.format.DateTimeFormatter;
import java.util.Date;

public class DateConverter {
    @TypeConverter
    public static Date fromTimestamp(Long value) {
        return value == null ? null : new Date(value);
    }

    @TypeConverter
    public static Long dateToTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }

    @SuppressLint("SimpleDateFormat")
    public static String toString(Date date){
        String pattern = "dd/MM/yyyy HH:mm:ss";
        return new SimpleDateFormat(pattern).format(date);
    }
}

File: app/src/main/java/com/luisenricke/simpleroomapp/data/converter/DateConverter.kt
package com.luisenricke.simpleroomapp.data.converter

import androidx.room.TypeConverter
import java.util.*

object DateConverter {
    @TypeConverter
    fun fromTimestamp(value: Long?): Date? = if (value == null) null else Date(value)

    @TypeConverter
    fun dateToTimestamp(date: Date?): Long? = date?.time
}